### PR TITLE
Adding delete collection

### LINKED ISSUE #22: Delete a Storage Collection
Ability to DELETE a storage collection

Acceptance Criteria:

- returns 404 when no record is found
- 400 when trying to delete a storage root
-  cannot delete when the item has children - 400
- 400 when it's not a storage collection
- 3 400 responses - should send back messages
- returns 204 when successful
- removed from database
- make problem responses discernible (via problem code/error etc)

### PR DESCRIPTION
**Pull request by JackLewis-digirati**
Resolves #22

This PR adds the ability to delete a collection

The format for this request is `DELETE {base URL}/{customer id}/collections/{collection id}`

In order to perform a delete, the request must be authorized and have a `X-IIIF-CS-Show-Extras` header

### CODE REVIEW

File: src/IIIFPresentation/API/Features/Storage/Requests/DeleteCollection.cs [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+        if (collection is null) return new ResultMessage<DeleteResult, DeleteCollectionType>(DeleteResult.NotFound);
+
+        if (collection.Parent is null)
+        {
+            return new ResultMessage<DeleteResult, DeleteCollectionType>(DeleteResult.BadRequest,
+                DeleteCollectionType.CannotDeleteRootCollection, "Cannot delete a root collection");
+        }
+
+        var itemCount = await dbContext.Collections.CountAsync(
```
Comment (donaldgray) [inline src/IIIFPresentation/API/Features/Storage/Requests/DeleteCollection.cs:45]: Is `.Any()` quicker than `.Count()`? IIRC `.Any()` will use `EXISTS` in postgres, which stops if it finds a match, whereas `COUNT` will count all matches.

File: src/IIIFPresentation/API/Features/Storage/Requests/DeleteCollection.cs [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+}
+
+public class DeleteCollectionHandler(
+    PresentationContext dbContext,
+    ILogger<CreateCollection> logger)
+    : IRequestHandler<DeleteCollection, ResultMessage<DeleteResult, DeleteCollectionType>>
+{
+    private const string RootCollection = "root";
+    
+    public async Task<ResultMessage<DeleteResult, DeleteCollectionType>> Handle(DeleteCollection request, CancellationToken cancellationToken)
```
Comment (donaldgray) [inline src/IIIFPresentation/API/Features/Storage/Requests/DeleteCollection.cs:23]: Is it worth adding customer to any of these logs? Would it help later?

File: src/IIIFPresentation/API.Tests/Integration/ModifyCollectionTests.cs [hunk 3]
```diff
@@ -464,4 +464,114 @@ public async Task UpdateCollection_FailsToUpdateCollection_WhenCalledWithoutNeed
         // Assert
         response.StatusCode.Should().Be(HttpStatusCode.Forbidden);
     }
+    
+    [Fact]
+    public async Task DeleteCollection_DeletesCollection_WhenAllValuesProvided()
```
Comment (donaldgray) [inline src/IIIFPresentation/API.Tests/Integration/ModifyCollectionTests.cs:469]: What does `WhenAllValuesProvider` mean here?

**Review by donaldgray — APPROVED** [review]
LGTM, couple of minor comments. Also, I'm not sure but should `ETag` be required for deletion?

File: src/IIIFPresentation/API/Features/Storage/Requests/DeleteCollection.cs [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+
+        var itemCount = await dbContext.Collections.CountAsync(
+            c => c.CustomerId == request.CustomerId && c.Parent == collection.Id,
+            cancellationToken: cancellationToken);
+
+        if (itemCount != 0)
+        {
+            return new ResultMessage<DeleteResult, DeleteCollectionType>(DeleteResult.BadRequest,
+                DeleteCollectionType.CollectionNotEmpty, "Cannot delete a collection with child items");
+        }
```
Comment (stephenwf) [inline src/IIIFPresentation/API/Features/Storage/Requests/DeleteCollection.cs:53]: Can collections be circular?

File: src/IIIFPresentation/API.Tests/Integration/ModifyCollectionTests.cs (hunk 3 above)
Comment (JackLewis-digirati) [reply src/IIIFPresentation/API.Tests/Integration/ModifyCollectionTests.cs:469]: `WhenAllValuesProvided` is meaning that everything has been provided that's needed to delete something successfully, like id, headers, etc

**Pull request merged**